core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -751,10 +750,14 @@ private SqlNode createLeftCall(SqlOperator op, List<SqlNode> nodeList) {
       if (nodeList.size() == 2) {
```

Comment:
I think we may also drop this "== 2" specialization as the method is no longer recursive
+1

Reply:
updated, thanks!